### PR TITLE
PE-635: Adds a stronger type than any and fixes the emerged errors/warnings 

### PR DESCRIPTION
**Pull request by matibat**

### CODE REVIEW

File: src/CLICommand/parameters_helper.ts [hunk 1]
```diff
@@ -225,10 +225,21 @@ export class ParametersHelper {
...
+		// FIXME: it could also return an array or a boolean!
 		const value = this.options[parameterName];
 		if (!value) {
 			throw new Error(`Required parameter ${parameterName} wasn't provided!`);
 		}
 		return value;
 	}
+
+	// eslint-disable-next-line @typescript-eslint/no-explicit-any
+	public getRequiredParameterValueTyped<T>(parameterName: ParameterName, closure: (a: any) => T): T {
```
[the remark this answers is not in the document]
Comment (matibat) [reply src/CLICommand/parameters_helper.ts:237]: Actually, that's a good one! Lemme try

[the remark this answers is not in the document]
Comment (matibat) [reply src/CLICommand/parameters_helper.ts:237]: It has conflicts with the generic `T`: `'T' could be instantiated with an arbitrary type which could be unrelated to 'string'.`

[the remark this answers is not in the document]
Comment (matibat) [reply src/CLICommand/parameters_helper.ts]: See here: 6b5ab71

File: src/CLICommand/parameters_helper.ts [hunk 2]
```diff
@@ -224,11 +224,16 @@ export class ParametersHelper {
 	 * @returns {string | undefined}
 	 * Returns the string value for the specific parameter; throws an error if not set
 	 */
-	public getRequiredParameterValue(parameterName: ParameterName): string {
+	public getRequiredParameterValue<T = string>(
+		parameterName: ParameterName,
+		// eslint-disable-next-line @typescript-eslint/no-explicit-any
+		closure: (input: any) => T = (input: any) => input as T
+	): T {
+		// FIXME: it could also return an array or a boolean!
```
Comment (matibat) [inline src/CLICommand/parameters_helper.ts:232]: We will surely implement different methods for reading boolean & array parameters. So this function may throw when reading a boolean/array

File: src/CLICommand/parameters_helper.ts [hunk 3]
```diff
@@ -224,11 +224,16 @@ export class ParametersHelper {
 	 * @returns {string | undefined}
 	 * Returns the string value for the specific parameter; throws an error if not set
```
Comment (matibat) [inline src/CLICommand/parameters_helper.ts:225]:
```suggestion
	 * @param {(input: any) => T} - A closure that casts the primitive value into a safe type
	 * @throws - When the required parameter value has a falsy value
	 * Returns the string value for the specific parameter
```

**Review by arielmelendez — APPROVED** [review]
Will merge after necessary fixes on dest branch.

**Pull request merged**